### PR TITLE
Display owner playlists

### PR DESCRIPTION
**Pull request by shellygamlielly**
1. Use Axios instead of fetch
2. Add Playlists and Song component to support displaying the user playlists
3. SpotifyLoginSuccess navigates to Welcome if the user does not exist else to the home page.
4. If the user exists the home page should request the userId by the spotifyId. For now, each time the user is requested to log in with Spotify.

### CODE REVIEW

File: src/constants/playlist.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export interface Playlist {
```
Comment (ramiloif) [inline src/constants/playlist.ts:1]: 1.til
2. DTO folder and  add Dto to class name

File: src/Playlists.tsx [hunk 2]
```diff
@@ -1,5 +1,93 @@
...
+
+  const ownerId = context.userId;
+
+  useEffect(() => {
+    const fetchPlaylists = async () => {
+      try {
+        const response = await axios.get(
+          `http://localhost:3000/playlist/${ownerId}`,
+        );
+        const playlistsData = response.data.map((playlist: any) => ({
```
Comment (ramiloif) [inline src/Playlists.tsx:39]: delete all the map here

File: src/Playlists.tsx [hunk 3]
```diff
@@ -1,5 +1,93 @@
...
+        console.error("Error fetching userId:", error);
+      }
+    };
+
+    if (!context.userId) {
+      fetchUserId();
+    }
+  }, [context, spotifyId]);
+
+  const ownerId = context.userId;
```
Comment (ramiloif) [inline src/Playlists.tsx:31]: not need this var just use user id

File: src/Playlists.tsx [hunk 4]
```diff
@@ -1,5 +1,93 @@
...
+interface PlaylistsProps {
+  spotifyId: string;
+}
+
+function Playlists({ spotifyId }: PlaylistsProps) {
+  const [playlists, setPlaylists] = useState<Playlist[]>([]);
+  const context = useUserId();
+
+  useEffect(() => {
+    const fetchUserId = async () => {
```
Comment (ramiloif) [inline src/Playlists.tsx:15]: This should be done via context in login-success

File: src/Playlists.tsx [hunk 5]
```diff
@@ -1,5 +1,93 @@
...
+  const handleCreatePlaylist = async () => {
+    const playlistName = prompt("Enter playlist name:");
+    if (playlistName && ownerId) {
+      await createPlaylist(playlistName, ownerId);
+    }
+  };
+
+  async function createPlaylist(name: string, ownerId: string) {
+    try {
+      const playlistId = await axios.post(`http://localhost:3000/playlist/`, {
```
Comment (ramiloif) [inline src/Playlists.tsx:67]: after sucess refetch

File: src/Playlists.tsx [hunk 6]
```diff
@@ -1,5 +1,93 @@
...
+
+    if (!context.userId) {
+      fetchUserId();
+    }
+  }, [context, spotifyId]);
+
+  const ownerId = context.userId;
+
+  useEffect(() => {
+    const fetchPlaylists = async () => {
```
Comment (ramiloif) [inline src/Playlists.tsx:34]: exctract this method out

File: src/spotify-login-success.tsx [hunk 7]
```diff
@@ -22,7 +24,10 @@ function SpotifyLoginSuccess() {
         if (!result.error) {
           setSpotifyProfile(result);
           // todo: use access token instead of user id
-          setTunityProfile(await getUser(result.id));
+          const user = await getUser(result.id);
+          if (user) {
+            setTunityProfile(user);
```
Comment (ramiloif) [inline src/spotify-login-success.tsx:29]: Instead of this use navigate

**Review by ramiloif — COMMENTED** [review]
Bonus what will happened if the user will go directly to `/playlist` via his browser?

File: src/Playlists.tsx (hunk 5 above)
Comment (shellygamlielly) [reply src/Playlists.tsx:67]: i extracted the method fetchPlaylists and called it after creating a new playlist

File: src/Playlists.tsx (hunk 4 above)
Comment (shellygamlielly) [reply src/Playlists.tsx:15]: correct also context has more information about the user now.

**Pull request closed without merging**